wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceRuntimes.java

```diff
@@ -340,9 +340,11 @@ public WorkspaceStatus getStatus(String workspaceId) {
     } catch (ValidationException e) {
       LOG.error(e.getLocalizedMessage(), e);
       throw new ConflictException(e.getLocalizedMessage());
-    } catch (InfrastructureException e) {
+    } catch (InternalInfrastructureException e) {
       LOG.error(e.getLocalizedMessage(), e);
       throw new ServerException(e.getLocalizedMessage(), e);
+    } catch (InfrastructureException e) {
+      throw new ServerException(e.getLocalizedMessage(), e);
```

Comment:
Shouldn't we review such cases and add a comment in the code to prevent the appearance of the same discussion in future?
I don't expect debug logs to be reviewed if they are not collected by the error tracking system, so I would either apply this PR or close it.

Reply:
> Shouldn't we review such cases and add a comment in the code to prevent the appearance of the same discussion in future?

I afraid sometimes it is hardly possible to check if an exception is related to system state or user wrong operation. 
For example a case with remapping `KubernetesClientException` to `KubernetesInfrastructureException` that we do in many classes like `KubernetesDeployments`, `KubernetesNamespace`, etc. Actually, `KubernetesClientException` may be thrown if Che is configured with wrong credentials, OCP instance is unavailable, or when a user doesn't have enough resources, send incorrect data (like wrong name format or object with the same name already exist).